File: src/Blockchain.java
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Blockchain {
    private List<Block> chain;

    public Blockchain() {
        this.chain = new ArrayList<>();
        Transaction t = new Transaction("system", "system", 0);
        Block genesis = new Block(null, "genesis hash", t, LocalDateTime.now(), 3, 0);
        // mejor hacer un metodo statico genesis() en block y que devuelva el bloque genesis
        //chain.add(Block.genesis)
        chain.add(genesis);
    }

    public void addBlock(Transaction transaction) {
        Block newBlock = Block.mine(this.getLatestBlock(), transaction);
        chain.add(newBlock);
    }

    public Block getLatestBlock() {
        return chain.get(chain.size() - 1);
    }

    @Override
    public String toString() {
        return "Blockchain{" +
                "chain=" + chain +
                '}';
    }
}

File: blockchain/Blockchain.py
from blockchain.Block import *
from blockchain.Transaction import *

class Blockchain:
    def __init__(self):
        self.chain = []
        # print(self.chain)

    def add_block(self, transaction):
        new_block = Block.mine(self.get_latest_block(), transaction)
        self.chain.append(new_block)

    def get_latest_block(self):
        return self.chain[-1]

    def get_chain(self):
        return self.chain

    def __str__(self):
        return f"Blockchain = {self.chain}"


